### ISSUE TITLE
XmlMapper.copy() doesn't properly copy internal configurations

### ISSUE THREAD

**Issue by gtrog** [FasterXML/jackson-dataformat-xml#233]
Affected version: all versions (up to jackson-dataformat-xml-2.9.0.pr3)

Currently, if you create an instance of XmlMapper and register a module with it, any copies of that mapper will not contain the registered module.  This behavior is different from the behavior of `ObjectMapper`, which retains all other configurations, such as modules, serialization/deserialization configs, etc.

```
final XmlMapper mapper = new XmlMapper();
mapper.registerModule(new JaxbAnnotationModule());

final XmlMapper mapperWithoutModule = mapper.copy();
```

I think this could be easily resolved by following the same copy-constructor pattern that `ObjectMapper` uses to copy itself, and then just invoking super's constructor in `XmlMapper`'s copy constructor.

I'll have a PR up shortly.

**Comment by cowtowncoder** [comment]
Thank you for reporting this: yes, copy() should retain module settings.

I added one important comment; I think serializer provider does need to be copied. Otherwise LGTM.

**Comment by cowtowncoder** [comment]
Thank you! I also backported this in 2.8, to be included in 2.8.9.
